tests/core/mempool/test_mempool_manager.py

```diff
@@ -136,7 +178,92 @@ async def test_block_cost_exceeds_max() -> None:
 async def test_double_spend_prevalidation() -> None:
     mempool_manager = await instantiate_mempool_manager(zero_calls_get_coin_record)
     conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
-    sb = spend_bundle_from_conditions(conditions)
+    sb = spend_bundle_from_conditions(TEST_COIN, conditions)
     sb_twice: SpendBundle = SpendBundle.aggregate([sb, sb])
     with pytest.raises(ValidationError, match="Err.DOUBLE_SPEND"):
         await mempool_manager.pre_validate_spendbundle(sb_twice, None, sb_twice.name())
+
+
+def test_run_for_cost_and_additions() -> None:
+    conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1]]
+    solution = Program.to(conditions)
+    with pytest.raises(ValueError, match="('cost exceeded', '01')"):
+        run_for_cost_and_additions(TEST_COIN_ID, IDENTITY_PUZZLE, solution, uint64(1))
+    created_coin = Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 1)
+    expected_cost, expected_coins = run_for_cost_and_additions(TEST_COIN_ID, IDENTITY_PUZZLE, solution, uint64(50))
+    assert expected_cost == 44
+    assert expected_coins == [created_coin]
+
+
+@pytest.mark.asyncio
+async def test_check_item_for_dedup() -> None:
```

Comment:
I think you should split this function up into a separate function for each case that you test

Reply:
Done.